### PR TITLE
build: add windows exe pack & upload CI job

### PR DESCRIPTION
**Pull request by RasPhilCo**
### What does this PR do?
adds windows pack and upload ci job

### What issues does this PR fix or reference?

[@W-8856338@](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B00000095vfXIAQ/view)

### CODE REVIEW

**Comment by RodEsp** [comment]
@RasPhilCo looks like you need to put a `@` next to the `W-8856338` to satisfy CI.

**Review by RodEsp — APPROVED** [review]

**Comment by RodEsp** [comment]
@RasPhilCo merge at your leisure. It doesn't seem I have permissions to do that yet.

**Comment by RasPhilCo** [comment]
Successful run: https://app.circleci.com/pipelines/github/salesforcecli/sfdx-cli/180/workflows/c218c8fb-40b2-4df3-93d7-3d2be6470cbf/jobs/789

File: .circleci/config.yml [hunk 1]
```diff
@@ -69,6 +69,19 @@ jobs:
       - run: yarn install
       - run: ./scripts/promote
 
+  pack-and-upload-windows-installer:
+    <<: *pack-and-upload-tarballs
```
Comment (RodEsp) [inline .circleci/config.yml:73]: TIL: [`<<` syntax in yaml](https://stackoverflow.com/questions/41063361/what-is-the-double-left-arrow-syntax-in-yaml-called-and-wheres-it-specced/41065222).

**Review by RodEsp — APPROVED** [review]

**Pull request merged**